libbeat/docs/upgrading.asciidoc

```diff
@@ -0,0 +1,203 @@
+(`C:\ProgramData\filebeat\registry`).
+
+If you are running Filebeat from the `tar.gz` archive, however, the location of
+the registry file has changed. In version 1.x it was by default named
+`.filebeat` in the current working directory. In version 5.0, it is
+`data/registry` relative to the binary location. For upgrading in this case, we
+recommend copying the `.filebeat` file to `data/registry`.
+
+==== Upgrading Topbeat to Metricbeat
+
+With the Beats 5.0 release, Topbeat is replaced by Metricbeat, which offers more
+functionality. More precisely, the Topbeat functionality is roughly equivalent
+with the "system" module in Metricbeat.
+
+While there is no automatic way to convert a Topbeat configuration file in the
+equivalent Metricbeat configuration file, both configuration are simple enough
+that migrating shouldn't take long.
+
+For example, this is the input section of the Topbeat configuration file:
+
+[source,yaml]
+------------------------------------------------------------------------------
+input:
+  # In seconds, defines how often to read server statistics
+  period: 10
+
+  # Regular expression to match the processes that are monitored
+  # By default, all the processes are monitored
+  procs: [".*"]
+
+  # Statistics to collect (all enabled by default)
+  stats:
+    # per system statistics, by default is true
+    system: true
+
+    # per process statistics, by default is true
+    process: true
+
+    # file system information, by default is true
+    filesystem: true
+
+    # cpu usage per core, by default is false
+    cpu_per_core: false
+------------------------------------------------------------------------------
+
+And here is the equivalent configuration for the system module in Metricbeat:
+
+[source,yaml]
+------------------------------------------------------------------------------
+metricbeat.modules:
+
+#------------------------------- System Module -------------------------------
+- module: system
+  metricsets:
+    # CPU stats
+    - cpu
+
+    # System Load stats
+    - load
+
+    # Per filesystem stats
+    - filesystem
+
+    # Memory stats
+    - memory
+
+    # Network stats
+    - network
+
+    # Per process stats
+    - process
+  enabled: true
+  period: 10s
+  processes: ['.*']
+------------------------------------------------------------------------------
+
+The rest of the configuration options (outputs, name, tags) stay the same or can
+be upgraded using the migration script.
+
+[[upgrade-mapping-template]]
+=== Upgrade the Elasticsearch mapping template
+
+By default, the Elasticsearch mapping template which comes with the Beats is not
+re-installed if it already exists.  This means that if the new Beats version
+adds new fields (e.g. a new module in Metricbeat), they won't be reflected in
+the Elasticsearch mapping, which can result in visualization problems or
+incorect data.
+
+You can set the `output.elasticsearch.template.overwrite` option to `true` to
```

Comment:
Should we add here an example with the -E option that this could be also done only  once on startup without having to modify the config file?